Failing on Android 7.1 (despite upgraded Chrome and WebView), but not because of this PR, there is an issue with the way notifications are broadcasted. It seems older Android versions were not tested since a long time. Here is the trace:

```
FATAL EXCEPTION: main
Process: com.superproductivity.superproductivity, PID: 9827
java.lang.RuntimeException: Error receiving broadcast Intent { act=com.superproductivity.superproductivity.UPDATE_PERMANENT_NOTIFICATION flg=0x10 (has extras) } in com.superproductivity.superproductivity.KeepAliveNotificationService$MyReceiver@c072e9d
    at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:1132)
    at android.os.Handler.handleCallback(Handler.java:751)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at android.os.Looper.loop(Looper.java:154)
    at android.app.ActivityThread.main(ActivityThread.java:6119)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
Caused by: kotlin.UninitializedPropertyAccessException: lateinit property notificationManager has not been initialized
    at com.superproductivity.superproductivity.KeepAliveNotificationService.updateNotification(KeepAliveNotificationService.kt:120)
    at com.superproductivity.superproductivity.KeepAliveNotificationService$MyReceiver.onReceive(KeepAliveNotificationService.kt:35)
    at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:1122)
    at android.os.Handler.handleCallback(Handler.java:751) 
    at android.os.Handler.dispatchMessage(Handler.java:95) 
    at android.os.Looper.loop(Looper.java:154) 
    at android.app.ActivityThread.main(ActivityThread.java:6119) 
    at java.lang.reflect.Method.invoke(Native Method) 
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886) 
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776) 
```

I'll see if I can fix that, but I'm not well versed into notifications.

--- a/app/src/main/java/com/superproductivity/superproductivity/CommonJavaScriptInterface.kt
+++ b/app/src/main/java/com/superproductivity/superproductivity/CommonJavaScriptInterface.kt
@@ -375,7 +375,12 @@ abstract class CommonJavaScriptInterface(
                 file?.openInputStream(activity)?.reader()
             } else {
                 // Older versions of Android <= 9 don't need scoped storage management
-                BufferedReader(FileReader(fullFilePath))
+                try {
+                    BufferedReader(FileReader(fullFilePath))
+                } catch (e: Exception) {
+                    // File does not exist, that's normal if it's the first time, we simply return null
+                    null
+                }
             }
 
         // Use a StringBuilder to rebuild the input file's content but replace the line returns with current OS's line returns
@@ -494,6 +499,7 @@ abstract class CommonJavaScriptInterface(
     @JavascriptInterface
     fun grantFilePermission(requestId: String) {
         // For Android < 10, ask for permission to access the whole storage
+        /* DEPRECATED: if we use this to get the permissions, then we need to use another folder picker than SimpleStorage, because otherwise SimpleStorage also asks for permissions, but Android does not accept asking for two different set of permissions in a single call: "Can reqeust only one set of permissions at a time"
         if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q) {
             ActivityCompat.requestPermissions(
                 activity, arrayOf(
@@ -502,6 +508,7 @@ abstract class CommonJavaScriptInterface(
                 ), 1
             )
         }
+        */
         // For Android >= 10, use scoped storage via SimpleStorage library to get the permission to write files in a folder
         // For Android < 10, SimpleStorage serves as a simple folder path picker, so that we still save where the user want look for a database file
         // Note that SimpleStorage takes care of all the gritty technical details, including whether the user must pick a root path BEFORE selecting the folder they want to store in, everything is explained to the user

--- a/app/src/main/java/com/superproductivity/superproductivity/KeepAliveNotificationService.kt
+++ b/app/src/main/java/com/superproductivity/superproductivity/KeepAliveNotificationService.kt
@@ -117,7 +117,9 @@ class KeepAliveNotificationService : Service() {
         if (message == "") {
             builder.setContentText(null)
         }
-        notificationManager.notify(NOTIFY_ID, builder.build())
+        if (::notificationManager.isInitialized) {
+            notificationManager.notify(NOTIFY_ID, builder.build())
+        }
     }
 
     private fun startForeground() {
@@ -131,8 +133,10 @@ class KeepAliveNotificationService : Service() {
                 importance
             )
             channel.description = description
-            notificationManager = getSystemService(NotificationManager::class.java)
-            notificationManager.createNotificationChannel(channel)
+            if (::notificationManager.isInitialized) {
+                notificationManager = getSystemService(NotificationManager::class.java)
+                notificationManager.createNotificationChannel(channel)
+            }
 
             // create notification
             val notificationIntent = Intent(this, FullscreenActivity::class.java)

--- a/app/src/main/java/com/superproductivity/superproductivity/KeepAliveNotificationService.kt
+++ b/app/src/main/java/com/superproductivity/superproductivity/KeepAliveNotificationService.kt
@@ -117,7 +117,7 @@ class KeepAliveNotificationService : Service() {
         if (message == "") {
             builder.setContentText(null)
         }
-        if (::notificationManager.isInitialized) {
+        if (::notificationManager.isInitialized) { // check if lateinit object is initialized, avoids Android <= 7.1 crashes
             notificationManager.notify(NOTIFY_ID, builder.build())
         }
     }
@@ -133,7 +133,7 @@ class KeepAliveNotificationService : Service() {
                 importance
             )
             channel.description = description
-            if (::notificationManager.isInitialized) {
+            if (::notificationManager.isInitialized) { // check if lateinit object is initialized, avoids Android <= 7.1 crashes
                 notificationManager = getSystemService(NotificationManager::class.java)
                 notificationManager.createNotificationChannel(channel)
             }

